### PR TITLE
Fix rpath-related Linux "test_shlibload" failure.

### PR DESCRIPTION
**Pull request by levitte**
When libssl and libcrypto are compiled on Linux with "-rpath", but
not "--enable-new-dtags", the RPATH takes precedence over
LD_LIBRARY_PATH, and we end up running with the wrong libraries.
This is resolved by using full (or at least relative, rather than
just the filename to be found on LD_LIBRARY_PATH) paths to the
shared objects.

-----

Alternative to #7596, on request

### CODE REVIEW

File: util/shlib_wrap.sh.in [hunk 1]
```diff
@@ -1,5 +1,22 @@
 #!/bin/sh
+{-
+    use lib '.';
+    use configdata;
 
+    sub shlib {
+        my $lib = shift;
+        return "" if $disabled{shared};
+        $lib = $unified_info{rename}->{$lib}
+            if defined $unified_info{rename}->{$lib};
```
Comment (levitte) [inline util/shlib_wrap.sh.in]: Note that strictly speaking, we don't currently need these rename rules on Unix.  However, should we *ever*...

**Comment by vdukhovni** [comment]
I tried your patch with 1.1.0 and it failed there, because $(SHLIB_MAJOR) etc. weren't defined in shlib_wrap.sh.  Perhaps will work in 1.1.1 and master, TBD (tomorrow).

**Comment by levitte** [comment]
I can create a separate 1.1.0 PR

**Comment by levitte** [comment]
Uhmmmmm...  I just cherry-picked to 1.1.0, and that was entirely unproblematic.  Considering I'm not touching `$(SHLIB_MAJOR)` anywhere, I'm not sure what happened in your case.

**Comment by vdukhovni** [comment]
> @levitte I just cherry-picked to 1.1.0, and that was entirely unproblematic.

I'll take a closer look to what went wrong, but it might be helpful if you would email me the generated `shlib_wrap.sh` for comparison.  I'm building on a Debian system, and "LIBSSLSO" and "LIBCRYPTOSO" ended up having unexpanded `${SHLIB_MAJOR}` and `${SHLIB_MINOR}' in their definitions, and:
```
make TESTS=test_shlibload V=1 test
```
failed. More details soon...

**Comment by levitte** [comment]
Ah!  Then I'm with you!  Ok, 1.1.0 special coming up.

**Comment by levitte** [comment]
#7631 is for 1.1.0

**Review by vdukhovni — APPROVED** [review]
This is also ready to go. Thanks!

**Comment by levitte** [comment]
It wasn't on Windows but is now.  Care to reconfirm?  Also #7631

File: Configurations/00-base-templates.conf [hunk 2]
```diff
@@ -102,6 +102,8 @@ my %targets=(
                                      ? "ranlib" : "" },
         RC              => "windres",
 
+        shared_extension => ".so",
+
```
Comment (vdukhovni) [inline Configurations/00-base-templates.conf]: Question: in `10-main.conf` for `darwin-common` I see:
```
shared_extension => ".\$(SHLIB_VERSION_NUMBER).dylib"
```
Which does not follow the "style" here, which looks at superficially inconsistent.  Is that a problem?

Comment (levitte) [reply Configurations/00-base-templates.conf]: That is perfectly correct, that's what Darwin shared library extensions looked like last I checked.  The "style" to follow is whatever is appropriate for the system the config target is for.

Comment (levitte) [reply Configurations/00-base-templates.conf]: s/extensions/"SONAME"/

**Review by vdukhovni — APPROVED** [review]
Just in case tested on Darwin, and it worked fine.
Re-confirming.

**Comment by levitte** [comment]
Thanks.  Merged

master:
45cdae1c7c93fe7ef2a981da4c36c3b8cb09e855 test/recipes/90-test_shlibload.t needs $target{shared_extension}
18289399743da6c3db462f37fc8797738e8acf7c Fix rpath-related Linux "test_shlibload" failure.

1.1.1:
c8bcea93b116d8d3dbae4fb32600a44e07b3c957 test/recipes/90-test_shlibload.t needs $target{shared_extension}
86736316b77e06fa9ed085a3e6fac3d337325ccb Fix rpath-related Linux "test_shlibload" failure.

**Pull request closed without merging**